File: app/models/trabajo_grado.py
from sqlalchemy import ForeignKey
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class TrabajoGrado(Base):
    __tablename__ = "trabajo_grado"

    id_trabajo: Mapped[int] = mapped_column(primary_key=True)
    id_proyecto: Mapped[str] = mapped_column(ForeignKey("proyecto.id_proyecto"))
    nombre: Mapped[str]
    facultad: Mapped[str]

File: flutter_app/lib/models/trabajo_grado.dart
class TrabajoGrado {
  final int idTrabajo;
  final String idProyecto;
  final String nombre;
  final String facultad;

  TrabajoGrado({
    required this.idTrabajo,
    required this.idProyecto,
    required this.nombre,
    required this.facultad,
  });

  factory TrabajoGrado.fromJson(Map<String, dynamic> json) {
    return TrabajoGrado(
      idTrabajo: json['id_trabajo'] as int,
      idProyecto: json['id_proyecto'] as String,
      nombre: json['nombre'] as String,
      facultad: json['facultad'] as String,
    );
  }
}
